cfme/tests/infrastructure/test_tenant_quota.py

```diff
@@ -132,3 +142,25 @@ def test_tenant_quota_enforce_via_service(request, appliance, provider, setup_pr
     def delete():
         provision_request.remove_request()
         catalog_item.delete()
+
+
+# first arg of parametrize is the list of fixtures or parameters,
+# second arg is a list of lists, with each one a test is to be generated
+# sequence is important here
+# indirect is the list where we define which fixtures are to be passed values indirectly.
+@pytest.mark.parametrize(
+    ['set_roottenant_quota', 'custom_prov_data'],
+    [
+        [('cpu', 2), {'change': 'cores_per_socket', 'value': 4}],
+        [('cpu', 2), {'change': 'sockets', 'value': 4}],
+        [('memory', 2), {'change': 'mem_size', 'value': 4096}]
+    ],
+    indirect=['set_roottenant_quota'],
+    ids=['max_cores', 'max_sockets', 'max_memory']
+)
+def test_tenant_quota_vm_reconfigure(appliance, provider, setup_provider, set_roottenant_quota,
+                                     small_vm, custom_prov_data):
+    new_config = small_vm.configuration.copy()
+    setattr(new_config.hw, custom_prov_data['change'], custom_prov_data['value'])
+    small_vm.reconfigure(new_config)
+    assert small_vm.configuration != new_config
```

Comment:
**Required** 

I'd like to see more verification here that the configuration has not changed. This only asserts it doesn't match the one that was attempted.

Save the original configuration and compare configuration after the reconfigure attempt to the original.

Reply:
Changed. Asserting to `original_config` now.